core/modules/arp_responder.cc

```diff
@@ -0,0 +1,79 @@
+#include "arp_responder.h"
+
+#include "../utils/arp.h"
+
+using bess::utils::Arp;
+using bess::utils::be16_t;
+
+const Commands ArpResponder::cmds = {
+    {"add", "ArpResponderArg", MODULE_CMD_FUNC(&ArpResponder::CommandAdd), 0}};
+
+CommandResponse ArpResponder::CommandAdd(const bess::pb::ArpResponderArg &arg) {
+  be32_t ip_addr;
+  arp_entry entry;
+
+  if (!arg.ip().length()) {
+    return CommandFailure(EINVAL, "IP address is missing");
+  }
+  if (!bess::utils::ParseIpv4Address(arg.ip(), &ip_addr)) {
+    return CommandFailure(EINVAL, "Invalid IP Address: %s", arg.ip().c_str());
+  }
+
+  if (!entry.mac_addr.FromString(arg.mac_addr())) {
+    return CommandFailure(EINVAL, "Invalid MAC Address: %s",
+                          arg.mac_addr().c_str());
+  }
+
+  entry.ip_addr = ip_addr;
+  entries_[ip_addr] = entry;
+  return CommandSuccess();
+}
+
+void ArpResponder::ProcessBatch(bess::PacketBatch *batch) {
+  gate_idx_t out_gates[bess::PacketBatch::kMaxBurst];
+  gate_idx_t incoming_gate = get_igate();
+
+  int cnt = batch->cnt();
+  for (int i = 0; i < cnt; i++) {
+    bess::Packet *pkt = batch->pkts()[i];
+
+    out_gates[i] = incoming_gate;
```

Comment:
Given that the module has only one input/output gate, you can just assign `0`. Unfortunately `get_igate()` is not free... :(